adal/util.py

```diff
@@ -55,7 +54,7 @@ def add_default_request_headers(self, options):
     headers[AdalIdParameters.SKU] = AdalIdParameters.PYTHON_SKU
     headers[AdalIdParameters.VERSION] = adal.__version__
     headers[AdalIdParameters.OS] = sys.platform
-    headers[AdalIdParameters.CPU] = 'x64' if platform.architecture()[0] == '64bit' else 'x86'
+    headers[AdalIdParameters.CPU] = 'x64' if sys.maxsize > 2 ** 32 else 'x86'

```

Comment:
Good point!

That `sys.maxsize > 2 ** 32` implementation is copied from [official document](https://docs.python.org/3/library/platform.html#cross-platform). We used it without too much thought. And yes, at a first look, it would be a bug on truely 32-bit platform.

But, as it turns out, Python supports long integer even on 32-bit platform. So, we do not really need to choose that exact biggest 32-bit number, we simply need any number larger than 2**31-1

```
>>> import platform
>>> platform.architecture()[0]
'32bit'
>>> import sys
>>> sys.maxint
2147483647
>>> 2**31-1
2147483647L
>>> 2**32
4294967296L
>>> sys.maxint > 2** 32
False
```

PS: I did not notice all these until you raised this question. Thank you. Learn something new each day!